src/test/java/de/qabel/desktop/DesktopClientGuiTest.java

```diff
@@ -0,0 +1,59 @@
+package de.qabel.desktop;
+
+import de.qabel.core.config.Account;
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DesktopClientGuiTest {
+
+    private TestKernel kernel;
+    private final AtomicBoolean exited = new AtomicBoolean(false);
+
+    @Test
+    public void notLoggedIn() throws Exception {
+        startTestKernel();
+        AsyncUtils.waitUntil(() -> kernel.app.config != null, 5000L);
+        kernel.app.config.onSetAccount(null);
+        Stage primaryStage = kernel.getContainer().getPrimaryStage();
+        AsyncUtils.assertAsync(primaryStage::isShowing);
+    }
+
+    @Test
+    public void loggedIn() throws Exception {
+        startTestKernel();
+        AsyncUtils.waitUntil(() -> kernel.app.config != null, 5000L);
+        kernel.app.config.onSetAccount(account -> new Account("provider", "user", "auth"));
```

Comment:
by calling `onSetAccount` you are registering an event handler. But I think what you actually want is triggering the event. That would be `setAccount(...)`